pkg/controllers/disruption/validation.go

```diff
@@ -165,15 +193,20 @@ func (e *EmptinessValidator) validateCandidates(ctx context.Context, candidates
 	}
 
 	if valid := lo.Filter(validatedCandidates, func(cn *Candidate, _ int) bool {
-		if e.cluster.IsNodeNominated(cn.ProviderID()) || disruptionBudgetMapping[cn.NodePool.Name] == 0 {
+		if e.cluster.IsNodeNominated(cn.ProviderID()) {
+			FailedValidationsTotal.Inc(map[string]string{ConsolidationTypeLabel: e.validationType, metrics.ReasonLabel: CandidatesIneligible})
+			return false
+		}
+		if disruptionBudgetMapping[cn.NodePool.Name] == 0 {
+			FailedValidationsTotal.Inc(map[string]string{ConsolidationTypeLabel: e.validationType, metrics.ReasonLabel: CandidatesIneligible})
 			return false
 		}
 		disruptionBudgetMapping[cn.NodePool.Name]--
 		return true
 	}); len(valid) > 0 {
 		return valid, nil
 	}
-	return nil, NewValidationError(fmt.Errorf("a candidate failed validation because it was nominated for a pod or would violate disruption budgets"))
+	return nil, NewValidationError(fmt.Errorf("%d candidates failed validation because it they were nominated for a pod or would violate disruption budgets", len(candidates)))
```

Comment:
```suggestion
	return nil, NewValidationError(fmt.Errorf("%d candidates failed validation because they were nominated for a pod or would violate disruption budgets", len(candidates)))
```

Reply:
Does it make sense to attach each candidate to a reason why it failed so that way we can count unique failures and return that in the error message